app/models/pacbio/library.rb

```diff
@@ -61,51 +59,31 @@ def create_used_aliquot
       )
     end
 
-    def create_tube
-      self.tube = tube || Tube.create!
-      save
-    end
-
-    def tube
-      pool ? pool.tube : super
-    end
-
     def collection?
       false
     end
 
+    # Note - This does not take into account when a library is used in a pool
+    # and that pool is used in a run
     # @return [Array] of Runs that the pool is used in
     def sequencing_runs
       wells&.collect(&:run)&.uniq
     end
 
+    # Note - This does not take into account when a library is used in a pool
+    # and that pool is used in a run
     # @return [Array] of Plates attached to a sequencing run
     def sequencing_plates
-      # TODO: remove this when pools are updated for aliquots
-      if pool
-        pool.sequencing_plates
-      else
-        wells&.collect(&:plate)
-      end
+      wells&.collect(&:plate)
     end
 
     private
 
-    def check_for_associated_wells
-      if wells.empty?
-        true
-      else
-        errors.add(:base, 'Cannot delete a library that is associated with wells')
-        throw(:abort)
-      end
-    end
-
-    def check_for_associated_pools
-      return true unless derived_aliquots.any? do |aliquot|
-                           aliquot.used_by_is_a?(Pacbio::Pool)
-                         end
+    # Derived aliquots indicate it has been used in a pool or run
+    def check_for_derived_aliquots
+      return true if derived_aliquots.empty?
```

Comment:
Is it possible to have a case where derived aliquots exists but it is not used by? I hope it is not and derived_aliquots will be cleaned once an existing used_by relation is removed.

Reply:
Good question. The derived aliquot should be deleted when the used_by container is deleted so this _**should**_ never happen.